cmd/juju/charmhub/download.go

```diff
@@ -133,6 +138,10 @@ func (c *downloadCommand) Init(args []string) error {
 		c.pipeToStdout = true
 	}
 
+	if c.pipeToStdout && c.resources && !c.noResources {
+		return errors.Errorf("cannot pipe to stdout and download resources: pass --no-resources to download to stdout")
+	}
```

Comment:
Can we remove the download option to stdout for 4.0 as well? Most will want to redirect it to a temp file, so we could do that for them.

Note: I'm a UNIX pipe output guy, but I'm thinking more of the CLI as a UX product that is focused and does one thing and one thing well.